### PR TITLE
Fix GIUH Logic for Tshirt C

### PR DESCRIPTION
**Pull request by robertbartel**
Fixing the logic for initializing the correct GIUH ordinates over the appropriate interpolation regularity in the logic for creating Tshirt_C realizations.

### CODE REVIEW

**Review by jdmattern-noaa — COMMENTED** [review]
Looks good to me.

**Comment by stcui007** [comment]
This PR produces the following runtime error:
$ ./cmake-build-dir/ngen ~/t-shirt/catchment_data.geojson cat-87
~/t-shirt/nexus_data.geojson nex-94 ~/t-shirt/realization_config.json
Hello there 0.1.0
Building Nexus collection
Building Catchment collection
terminate called after throwing an instance of 'std::runtime_error'
  what():  ERROR: No simulation time period defined.
Aborted (core dumped)

This might be inherited from an earlier merge.

File: src/core/catchment/giuh/GiuhJsonReader.cpp [hunk 1]
```diff
@@ -26,9 +26,23 @@ std::vector<double> GiuhJsonReader::extract_cumulative_frequency_ordinates(ptree
     std::vector<double> cumulative_freqs;
 
     // TODO: account for error condition of unmatching JSON structure for freqs
-    for (ptree::value_type freqs : catchment_data_node.get_child("CDF.CumulativeFreq")) {
+    std::string freq_node_name;
+    if (catchment_data_node.get_child_optional("CDF.CumulativeFreq") != boost::none) {
+        freq_node_name = "CDF.CumulativeFreq";
+    }
+    else {
+        freq_node_name = "CDF.minHydro.Runoff";
```
Comment (hellkite500) [inline src/core/catchment/giuh/GiuhJsonReader.cpp:34]: I don't believe this key in the GIUH is a cumulative freq, I think it is a computed incremental run off.  My suspicion is that if this else case is ever encountered, the runtime_error below will always trigger.

**Review by hellkite500 — CHANGES_REQUESTED** [review]

**Comment by hellkite500** [comment]
#144 is relevant here especially in understanding the keys used in GIUH.json.  Based on Brians comment in that issue, we definitely should remove the fallback to `CDF.minHydro.Runoff` in the frequency reading function.

File: src/core/catchment/giuh/GiuhJsonReader.cpp (hunk 1 above)
Comment (hellkite500) [reply src/core/catchment/giuh/GiuhJsonReader.cpp:34]: Just throw a runtime error here for now "GIUH doesn't contain CumlativeFreq key"

**Comment by hellkite500** [comment]
Merge once above comment is resolved

File: src/core/catchment/giuh/GiuhJsonReader.cpp (hunk 1 above)
Comment (robertbartel) [reply src/core/catchment/giuh/GiuhJsonReader.cpp]: Change applied.

**Review by hellkite500 — APPROVED** [review]

**Pull request merged**